### PR TITLE
Adding Flux RBAC in cluster ns

### PR DESCRIPTION
**Pull request by ljakimczuk**
## Description

Towards https://github.com/giantswarm/giantswarm/issues/21792.

## Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] If the change introduces new roles, please consider adding user-friendly descriptions to the roles with the annotation `giantswarm.io/notes` to help explain their purpose and usage.

## Test

Tested on `gauss`. Two additional `RoleBindings` have been created successfully and also removed upon cluster deletion. No `RoleBinding` has been created in unwanted namespace. No errors in the logs.

### CODE REVIEW

File: service/controller/clusternamespace/resource/clusternamespaceresources/create_test.go
Comment (MarcelMue) [inline service/controller/clusternamespace/resource/clusternamespaceresources/create_test.go:31]: Would be good to have another namespace to ensure we are not adding rolebindings to non-org namespaces.

File: service/controller/clusternamespace/resource/clusternamespaceresources/delete_test.go
Comment (MarcelMue) [inline service/controller/clusternamespace/resource/clusternamespaceresources/delete_test.go:105]: Same as above - not needed.

File: service/controller/clusternamespace/resource/clusternamespaceresources/delete_test.go
Comment (MarcelMue) [inline service/controller/clusternamespace/resource/clusternamespaceresources/delete_test.go:88]: Same here - error check missing.

File: service/controller/clusternamespace/resource/clusternamespaceresources/create_test.go
Comment (ljakimczuk) [reply service/controller/clusternamespace/resource/clusternamespaceresources/create_test.go:31]: good point, will add it as well

File: pkg/key/key.go
Comment (kubasobon) [inline pkg/key/key.go:38]: Nice, glad it got moved to a common package.

File: service/controller/clusternamespace/resource/clusternamespaceresources/create_test.go
Comment (ljakimczuk) [reply service/controller/clusternamespace/resource/clusternamespaceresources/create_test.go:32]: @MarcelMue it's done, see the `expectedRoleBindingsNum` map

**Review by MarcelMue — APPROVED** [review]
To me this looks sane now - I guess Kuba has the best knowledge currently to tell if this is sensible as a whole.

**Review by anvddriesch — APPROVED** [review]
I haven't tested it but LGTM
Thanks for including tests :)

**Comment by ljakimczuk** [comment]
> I haven't tested it but LGTM

No worries, I will test it before merging.

**Review by kubasobon — APPROVED** [review]
Looks sane, but it's easy to miss something in review as RBAC is a complex beast.

LGTM when tested.

**Pull request merged**
